Shellcheck

BarbUk opened the pull request:
Fix several shellcheck warning and error

BarbUk committed:
remove trailling spaces

BarbUk committed:
useless cat

BarbUk committed:
SC2092: Remove backticks to avoid executing output.
SC2006: Use $(..) instead of legacy `..`.

BarbUk committed:
SC1102: Shells disambiguate $(( differently or not at all. For $(command substition), add space after $( .

BarbUk committed:
SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
SC1117: Backslash is literal in "\t". Prefer explicit escaping: "\\t".

BarbUk committed:
SC2166: Prefer [ p ] || [ q ] as [ p -o q ] is not well defined.

BarbUk committed:
indentation

BarbUk committed:
SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?

BarbUk committed:
SC2086: Double quote to prevent globbing and word splitting.
SC2004: $/${} is unnecessary on arithmetic variables.
SC2196: egrep is non-standard and deprecated. Use grep -E instead.

BarbUk committed:
SC2004: $/${} is unnecessary on arithmetic variables.

BarbUk committed:
indentation

BarbUk committed:
SC2017: Increase precision by replacing a/b*c with a*c/b.

BarbUk committed:
SC2154: critcalue is referenced but not assigned (did you mean 'critvalue'?).

BarbUk committed:
SC2196: egrep is non-standard and deprecated. Use grep -E instead.
SC2086: Double quote to prevent globbing and word splitting.

BarbUk commented:
The resulting script was tested on debian 9 with lxc 2

Napsty commented on check_lxc.sh:
```diff
@@ -51,19 +51,20 @@ sleep=5                 # define the sleep time between cpu checks
 if [[ $lxcversion -eq 0 ]]; then
-  for cmd in lxc-list; 
-  do if ! `which ${cmd} 1>/dev/null`
+  # shellcheck disable=2043
```
What do you mean by that?

Napsty commented:
Thanks @BarbUk 
These are a lot of speed improvements, learning a lot here ;-)
Would you be so kind and adapt the PR for the latest release (currently 0.6.2) to be able to merge? 
Thx

BarbUk committed:
Merge branch 'master' into fix/shellcheck

BarbUk committed:
SC2230: which is non-standard. Use builtin 'command -v' instead.

BarbUk committed:
SC2086: Double quote to prevent globbing and word splitting.

BarbUk committed:
fix grep usage

SC2143: Use grep -q instead of comparing output with [ -n .. ].
SC2143: Use ! grep -q instead of comparing output with [ -z .. ].

BarbUk commented:
Hey, thanks for the feedback.
I've merged the new version into my branch and resolved the conflicts.

I've added some new fix from shellcheck.
The resulting script has been tested against lxc 2.0.7-2 on a debian stretch.

Napsty commented:
Successfully tested on 

- Ubuntu 14.04 with LXC 1.0.8
- Ubuntu 18.04 with LXC 3.0.2 (however: #9 )

Thx for the contribution!
